tests/plugins/test_cancel_builds_on_autorebuild.py

```diff
@@ -0,0 +1,130 @@
+    prebuild_plugins = [{
+        'name': CancelBuildsOnAutoRebuild.key,
+        'args': {
+            "url": "",
+        }
+    }]
+
+    test_build_name = "testbuild-1"
+
+    def assert_message_logged(self, msg, cplog):
+        assert any([msg in l.getMessage() for l in cplog.records()])
+
+    def prepare(self):
+        if MOCK:
+            mock_docker()
+        tasker = DockerTasker()
+        workflow = DockerBuildWorkflow(SOURCE, "test-image")
+        setattr(workflow, 'builder', X())
+        setattr(workflow.builder, 'image_id', 'asd123')
+        setattr(workflow.builder, 'base_image', ImageName(repo='Fedora',
+                                                          tag='22'))
+        setattr(workflow.builder, 'source', X())
+        setattr(workflow.builder.source, 'path', '/tmp')
+        setattr(workflow.builder.source, 'dockerfile_path', None)
+        setattr(workflow, 'prebuild_results', {})
+
+        runner = PreBuildPluginsRunner(tasker, workflow, [
+            {
+                'name': CancelBuildsOnAutoRebuild.key,
+                'args': {
+                    'url': ''
+                }
+            }
+        ])
+        return workflow, runner
+
+    def test_without_rebuild(self, monkeypatch, caplog):
+        key = 'rebuild'
+        value = 'false'
+        buildconfig = "buildconfig1"
+
+        workflow, runner = self.prepare()
+        workflow.prebuild_results[CheckAndSetRebuildPlugin.key] = False
+
+        build_json = {
+            "metadata": {
+                "labels": {
+                    "buildconfig": buildconfig,
+                    key: value,
+                }
+            }
+        }
+
+        build = flexmock(
+            get_build_name=lambda: self.test_build_name,
+            is_running=lambda: False
+        )
+
+        monkeypatch.setenv("BUILD", json.dumps(build_json))
+        runner.run()
+        self.assert_message_logged(
+            "this is not an autorebuild, %s is doing nothing" % CancelBuildsOnAutoRebuild.key,
+            caplog
+        )
+
+    def test_with_rebuild(self, monkeypatch, caplog):
+        key = 'rebuild'
+        value = 'true'
+        workflow, runner = self.prepare()
+        workflow.prebuild_results[CheckAndSetRebuildPlugin.key] = True
+
+        monkeypatch.setenv("BUILD", json.dumps({
+            "metadata": {
+                "labels": {
+                    "buildconfig": "buildconfig1",
+                    key: value,
+                }
+            }
+        }))
+
+        build = flexmock(
+            get_build_name=lambda: self.test_build_name,
+            is_running=lambda: True,
+            build_id=self.test_build_name
+        )
+
+        atomic_reactor.plugins.pre_check_and_set_rebuild = flexmock(
+            is_rebuild=lambda x: True
+        )
+
+        flexmock(OSBS)
+        OSBS.should_receive("list_builds").and_return([build])
+        OSBS.should_receive("cancel_build").with_args(self.test_build_name)
```

Comment:
Also want to verify the situation where cancel_build raises an exception. Perhaps also use a list of builds?